torch/csrc/distributed/rpc/process_group_agent.h

```diff
@@ -161,12 +159,19 @@ class ProcessGroupAgent : public RpcAgent {
   MessageCounter recvCounts_;
 
   std::atomic<int64_t> nextId_;
+  // atomic bool indicating if this agent is running. It is set in
+  // ProcessGroupAgent::start and unset in ProcessGroupAgent::shutdown and
+  // ProcessGroupAgent::join. It controls whether several background threads
+  // should be running.
+  std::atomic<bool> rpcRunning_{false};
```

Comment:
Since you're locking access anyway and not using this as an atomic, I recommend changing this to a regular bool to avoid the impression you can use it as atomic.

Reply:
Synced with @xush6528 about this - currently, it is not always locked (see the checks in `listenLoop` and `send`). We only need to strictly lock it when it is used in conjunction with the `futureTimeout` condvar, so it might be worth leaving it as atomic?

Reply:
Sounds good, but please document what the expected locking strategy is.

Using an atomic implies (at least to me) that you don't need locks. To reduce the probability of this causing breakage later on, I think a regular bool is better. Accessing that without locks is a red flag in code review.

Reply:
Will add this in a follow up PR.

Reply:
Ended up doing it in this PR since had to do a rebase anyways.